man/measure.Rd

```diff
@@ -38,7 +38,10 @@ measurev(fun.list, sep, pattern, cols, multiple.keyword="value.name",
     are considered measure variables.}
   \item{pattern}{Perl-compatible regex with capture groups to match to
     \code{cols}. Columns that match the regex are considered measure variables.}
-  \item{cols}{A character vector of column names.}
+  \item{cols}{A character vector that is automatically provided, 
```

Comment:
I'm still not sure I understand (sorry, I'm not caught up yet on the new functionality of `melt()`). Can we simply remove this argument then? Or it's needed for internal calls to `melt()`? If that's the case, let's make that more evident?

```
\item{cols}{Do not use. This will be populated automatically with all of the column names of the data.table input to melt. To specify the columns to be measured, use the \code{pattern} argument.}
```

Reply:
we can not simply remove this argument, because it needs to be provided by eval_with_cols (for both measure and patterns).
I can update the docs to say "do not use" but based on your prior comments about maybe one day exporting this function, that may be confusing. What we really want is: users should not provide cols argument to measure or patterns, when they are used in the context of melt or [.data.table.

Reply:
couldn't we have 2 functions then, one internal and one meant to be used by users?

Reply:
I don't think so. I think we just need to document the NSE better, I will give it a try.